run/g4DetermineUnknownPdgProperties.cxx

```diff
@@ -120,8 +120,10 @@ int main(int argc, char** argv)
       continue;
     }
     auto name = particle->GetParticleName();
-    // but in the end we must add the actual - potentially isomeric - PDG
-    auto ret = pdgDB->AddParticle(name.c_str(), name.c_str(), particle->GetPDGMass(), particle->GetPDGStable(), particle->GetPDGWidth(), particle->GetPDGCharge(), "Ion", pdg);
+    // but in the end we must add the actual - potentially isomeric - PDG, make sure to divide mass by 1000 to go from G4 units (MeV) to ROOT units (GeV)
+    auto ret = pdgDB->AddParticle(name.c_str(), name.c_str(), particle->GetPDGMass() / 1000., particle->GetPDGStable(), particle->GetPDGWidth(), particle->GetPDGCharge(), "Ion", pdg);
+    // We also add the ground state (if that is the same as before, it will not be added anyways)
+    pdgDB->AddParticle(name.c_str(), name.c_str(), particle->GetPDGMass() / 1000., particle->GetPDGStable(), particle->GetPDGWidth(), particle->GetPDGCharge(), "Ion", pdgToLookFor);
```

Comment:
In Geant4 VMC this is implemented in [TG4ParticlesManager::AddParticleToPdgDatabase](https://github.com/vmc-project/geant4_vmc/blob/2a8ca929ed5ebf3ba68a98c78e8062c84125aacd/source/physics/src/TG4ParticlesManager.cxx#L97); the method is actually private so it cannot be used directly without changing the code.

Note, that  `particle->GetPDGWidth()` is also in energy units , so it needs to be converted; and the charge passed to TDatabasePDG has to be multiplied by `3` (this is a convention in Root). Also as you include Geant4 VMC classes, you can better use `TG4G3Units::Energy()` instead of 1000.

Reply:
Thanks a lot for your comments, I will take that into account for an update to this PR!

Reply:
Thanks for the update; the properties added to TDatabasePDG look fine now.